obspy/signal/array_analysis.py

```diff
@@ -21,16 +30,984 @@
+            self.inventory.plot(projection="local", show=False)
+            bmap = plt.gca().basemap
+
+            grav = self.center_of_gravity
+            x, y = bmap(grav["longitude"], grav["latitude"])
+            bmap.scatter(x, y, marker="x", c="red", s=40, zorder=20)
+            plt.text(x, y, "Center of Gravity", color="red")
+
+            geo = self.geometrical_center
+            x, y = bmap(geo["longitude"], geo["latitude"])
+            bmap.scatter(x, y, marker="x", c="green", s=40, zorder=20)
+            plt.text(x, y, "Geometrical Center", color="green")
+
+            plt.show()
+
+    def _get_geometry(self):
+        if not self.inventory:
+            return {}
+
+        geo = {}
+        for network in self.inventory:
+            for station in network:
+                station_code = "{n}.{s}".format(n=network.code,
+                                                s=station.code)
+                # todo this must work without channels, for just stations!
+                for channel in station:
+                    this_coordinates = \
+                        {"latitude": float(channel.latitude),
+                         "longitude": float(channel.longitude),
+                         "absolute_height_in_km":
+                             float(channel.elevation - channel.depth) / 1000.0}
+                    if station_code in geo and \
+                                    this_coordinates not in list(geo.values()):
+                        msg = ("Different coordinates for station '{n}.{s}' "
+                               "in the inventory. The first ones encountered "
+                               "will be chosen.".format(n=network.code,
+                                                        s=station.code))
+                        warnings.warn(msg)
+                        continue
+                    geo[station_code] = this_coordinates
+        return geo
+
+    @property
+    def geometrical_center(self):
+        extend = self.extend
+        return {
+            "latitude": (extend["max_latitude"] +
+                         extend["min_latitude"]) / 2.0,
+            "longitude": (extend["max_longitude"] +
+                          extend["min_longitude"]) / 2.0,
+            "absolute_height_in_km":
+                (extend["min_absolute_height_in_km"] +
+                 extend["max_absolute_height_in_km"]) / 2.0
+        }
+
+    @property
+    def center_of_gravity(self):
+        lats, lngs, hgts = self.__coordinate_values()
+        return {
+            "latitude": np.mean(lats),
+            "longitude": np.mean(lngs),
+            "absolute_height_in_km": np.mean(hgts)}
+
+    @property
+    def geometry(self):
+        return self._get_geometry()
+
+    @property
+    def aperture(self):
+        """
+        The aperture of the array in kilometers.
+        """
+        distances = []
+        geo = self.geometry
+        for station, coordinates in list(geo.items()):
```

Comment:
It's not necessary to add `list` when you're just going to iterate through the items.